core/tee/tee_fs.c

```diff
@@ -37,447 +37,56 @@
 #include <trace.h>
 #include <kernel/handle.h>
 
-/* TEE FS operation */
-#define TEE_FS_OPEN       1
-#define TEE_FS_CLOSE      2
-#define TEE_FS_READ       3
-#define TEE_FS_WRITE      4
-#define TEE_FS_SEEK       5
-#define TEE_FS_UNLINK     6
-#define TEE_FS_RENAME     7
-#define TEE_FS_TRUNC      8
-#define TEE_FS_MKDIR      9
-#define TEE_FS_OPENDIR   10
-#define TEE_FS_CLOSEDIR  11
-#define TEE_FS_READDIR   12
-#define TEE_FS_RMDIR     13
-#define TEE_FS_ACCESS    14
-
-struct tee_fs_fd {
-	int nw_fd;		/* normal world fd */
-	uint32_t flags;
-	uint32_t fp;		/* file pointer offset */
-	uint32_t len;
-};
-
-struct tee_fs_dir {
-	int nw_dir;
-	struct tee_fs_dirent d;
-};
-
-static struct handle_db fs_handle_db = HANDLE_DB_INITIALIZER;
-
-static int tee_fs_open(const char *file, int flags, ...)
-{
-	int res = -1;
-	size_t len;
-	struct tee_fs_fd *fd = NULL;
-	struct tee_fs_rpc head = { 0 };
-
-	len = strlen(file) + 1;
-	if (len > TEE_FS_NAME_MAX)
-		goto exit;
-
-	/* fill in parameters */
-	head.op = TEE_FS_OPEN;
-	head.flags = flags;
-	head.fd = 0;
-
-	res = tee_fs_send_cmd(&head, (void *)file, len, TEE_FS_MODE_IN);
-	if (res)
-		goto exit;
-
-	res = head.res;
-
-	if (res == -1)
-		goto exit;
-
-	fd = (struct tee_fs_fd *)malloc(sizeof(struct tee_fs_fd));
-	if (fd == NULL)
-		goto exit;
-
-	/* init internal status */
-	fd->nw_fd = head.fd;
-	fd->flags = flags;
-
-	/* return fd */
-	res = handle_get(&fs_handle_db, fd);
-
-exit:
-	if (res == -1)
-		free(fd);
-
-	return res;
-}
+#include "tee_fs_private.h"
 
 static int tee_fs_close(int fd)
 {
-	int res = -1;
-	struct tee_fs_fd *fdp = handle_put(&fs_handle_db, fd);
-	struct tee_fs_rpc head = { 0 };
-
-	if (fdp == NULL)
-		return -1;
-
-	/* fill in parameters */
-	head.op = TEE_FS_CLOSE;
-	head.fd = fdp->nw_fd;
-
-	res = tee_fs_send_cmd(&head, NULL, 0, TEE_FS_MODE_NONE);
-	if (res)
-		goto exit;
-
-	res = head.res;
+	struct tee_fs_fd *fdp = tee_fs_fd_put(fd);
```

Comment:
Would prefer to have `tee_fs_fd_put` part of `tee_fs_common_close`, so that the code is symmetric between the put and the get (all are done in common code).

I guess that `tee_fs_common_close` can be called directly in this case